### PR TITLE
Disables text editor for DMs and GMs with remote users

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
This change makes the text editor behave similarly to the case of read only channels, in a disabled state and showing a placeholder that indicates that DMs and GMs with remote users are currently not supported.

There is no way to create this type of channels now, but chances are that existing servers have some already, so this is intended to explicitly indicate that those won't work.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-60971

#### Screenshots
image

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by lindy65 — APPROVED** [review]
QA-deferring the PR to test post merge @mgdelacroix

**Review by sbishel — APPROVED** [review]
Nice!

File: webapp/channels/src/components/advanced_text_editor/advanced_text_editor.tsx [hunk 1]
```diff
@@ -109,6 +109,8 @@ const AdvancedTextEditor = ({
 
     const currentUserId = useSelector(getCurrentUserId);
     const channelDisplayName = useSelector((state: GlobalState) => getChannelSelector(state, channelId)?.display_name || '');
+    const channelType = useSelector((state: GlobalState) => getChannelSelector(state, channelId)?.type || '');
+    const isChannelShared = useSelector((state: GlobalState) => getChannelSelector(state, channelId)?.shared);
```
Comment (sbishel) [inline webapp/channels/src/components/advanced_text_editor/advanced_text_editor.tsx:113]: One thing...should we just call `getChannelSelector` once?

Comment (calebroseland) [reply webapp/channels/src/components/advanced_text_editor/advanced_text_editor.tsx:113]: +1

**Review by calebroseland — APPROVED** [review]

File: webapp/channels/src/components/advanced_text_editor/advanced_text_editor.tsx (hunk 1 above)
Comment (mgdelacroix) [reply webapp/channels/src/components/advanced_text_editor/advanced_text_editor.tsx]: Fixed on 04f11e1, PTAL

**Review by calebroseland — APPROVED** [review]

**Pull request merged**